lib/katello/engine.rb

```diff
@@ -195,6 +195,15 @@ def find_assets(args = {})
       ::Api::V2::HostsController.send :include, Katello::Concerns::Api::V2::HostsControllerExtensions
       ::Api::V2::HostgroupsController.send :include, Katello::Concerns::Api::V2::HostgroupsControllerExtensions
 
+      #facet extensions
+      Facets.register(Katello::Host::ContentFacet, :content_facet) do
+        api_view :list => 'katello/api/v2/content_facet/base_with_root', :single => 'katello/api/v2/content_facet/show'
+      end
+
+      Facets.register(Katello::Host::SubscriptionFacet, :subscription_facet) do
+        api_view :list => 'katello/api/v2/subscription_facet/base_with_root', :single => 'katello/api/v2/subscription_facet/show'
+      end
```

Comment:
I think what I mean is that  the plugin API would have contained the registration "code". In other words, you'd use the plugin API to register your facets instead of using the facets API so that the API plugin remains the well defined place to add your bits. If I am way off base, just tell me, if I am on to something I am more than happy to own a ticket.